interactive_engine/src/gaia-adaptor/src/main/java/com/alibaba/graphscope/gaia/MaxGraphStore.java

```diff
@@ -0,0 +1,68 @@
+package com.alibaba.graphscope.gaia;
+
+import com.alibaba.graphscope.gaia.store.GraphStoreService;
+import com.alibaba.graphscope.gaia.store.SchemaNotFoundException;
+import com.alibaba.maxgraph.compiler.api.schema.GraphSchema;
+import com.alibaba.maxgraph.compiler.api.schema.SchemaFetcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MaxGraphStore extends GraphStoreService {
+    private static final Logger logger = LoggerFactory.getLogger(MaxGraphStore.class);
+    public static final String MAXGRAPH_MODERN_PROPERTY_RESOURCE = "maxgraph.modern.properties.json";
+    private CachedGraphSchemaPair cachedGraphSchemaPair;
+
+    public MaxGraphStore(SchemaFetcher schemaFetcher) {
+        super(MAXGRAPH_MODERN_PROPERTY_RESOURCE);
+        this.cachedGraphSchemaPair = new CachedGraphSchemaPair(schemaFetcher);
+    }
+
+    @Override
+    public long getLabelId(String label) {
+        try {
+            GraphSchema graphSchema = this.cachedGraphSchemaPair.get().getLeft();
+            return graphSchema.getElement(label).getLabelId();
+        } catch (Exception e) {
+            logger.error("label " + label + " is invalid, please check schema");
+            throw new SchemaNotFoundException("label " + label + " is invalid, please check schema");
+        }
+    }
+
+    @Override
+    public String getLabel(long labelId) {
+        GraphSchema graphSchema = this.cachedGraphSchemaPair.get().getLeft();
+        return graphSchema.getElement((int) labelId).getLabel();
+    }
+
+    @Override
+    public long getGlobalId(long labelId, long propertyId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int getPropertyId(String propertyName) {
+        try {
+            GraphSchema graphSchema = this.cachedGraphSchemaPair.get().getLeft();
+            return graphSchema.getPropertyId(propertyName);
+        } catch (Exception e) {
+            logger.error("property " + propertyName + " is invalid, please check schema");
+            throw new SchemaNotFoundException("property " + propertyName + " is invalid, please check schema");
+        }
+    }
+
+    @Override
+    public String getPropertyName(int propertyId) {
+        GraphSchema graphSchema = this.cachedGraphSchemaPair.get().getLeft();
+        return graphSchema.getPropertyName(propertyId);
+    }
+
+    @Override
+    public long getSnapShotId() {
+        return this.cachedGraphSchemaPair.get().getRight();
+    }
+
+    @Override
+    public synchronized void updateSnapShotId() {
```

Comment:
There is a concurrent bug. If a query call this method while another query is processing, they may use two different schema. Open an issue and fix it later.